cmd/frontend/graphqlbackend/search_results.go

```
@@ -1217,6 +1220,36 @@ func (r *searchResolver) resultsWithTimeoutSuggestion(ctx context.Context) (*Sea
 		suggestTime := longer(2, usedTime)
 		return alertForTimeout(usedTime, suggestTime, r).wrap(), nil
 	}
+
+	if rr == nil && err != nil {
+		return rr, err
+	}
+
+	// rr != nil
+	var alert *searchAlert
+	newAlert := alertForDiffCommitSearch(err)
+	if newAlert != nil {
+		alert = newAlert
+	}
+	newAlert = alertForStructuralSearch(err)
+	if newAlert != nil {
+		alert = newAlert
+	}
+	if len(rr.SearchResults) == 0 && r.patternType != query.SearchTypeStructural && matchHoleRegexp.MatchString(r.originalQuery) {
+		alert = alertForStructuralSearchNotSet(r.originalQuery)
+	}
+	missingRepoRevs := errMissingRepoRevs{}
+	if errors.As(err, &missingRepoRevs) {
+		alert = alertForMissingRepoRevs(missingRepoRevs.patternType, missingRepoRevs.missingRepoRevs)
+	}
+	// If we have some results or an alert, only log the error instead of returning it.
+	// Otherwise the client would not receive the partial results or see the alert.
+	if (len(rr.SearchResults) > 0 || alert != nil) && err != nil {
+		log15.Error("Errors during search", "error", err)
+		err = nil
```

As per above, I'm concerned that `err` here could be a super bad thing, and we will be returning invalid partial results/alerts when it really is a bad error. This decision seems to hinge on:

"if we have partial results or an alert, but also an error, suppress the internal error, whatever it may be". 

I'm worried that will bite us (I, personally, am not convinced that the kinds of errors we could get here imply they can be suppressed if there are partial results/alerts, because I don't know what kinds of errors those could be). I would feel a lot better if there was, say, a switch statement in here that cased out on all the values that `err` could be, and decided whether it was OK to suppress `err`. If that seems like something we can't do soon, I would prefer we stick with the existing behavior until we can do that.

---

I switched back to something that is very close to the original logic. We ignore errors only if we have found results. If we haven't found any results and encountered an unhandled error, we surface the error.